src/script/lua_api/l_inventory.cpp

```diff
@@ -264,19 +264,19 @@ int InvRef::l_set_lists(lua_State *L)
 	}
 
 	// Make a temporary inventory in case reading fails
-	Inventory *tempInv(inv);
-	tempInv->clear();
+	Inventory tempInv(*inv);
+	tempInv.clear();
 
 	Server *server = getServer(L);
 
 	lua_pushnil(L);
 	luaL_checktype(L, 2, LUA_TTABLE);
 	while (lua_next(L, 2)) {
-		const char *listname = lua_tostring(L, -2);
-		read_inventory_list(L, -1, tempInv, listname, server);
+		const char *listname = luaL_checkstring(L, -2);
+		read_inventory_list(L, -1, &tempInv, listname, server);
 		lua_pop(L, 1);
 	}
-	inv = tempInv;
+	*inv = tempInv;
```

Comment:
```suggestion
	*inv = std::move(tempInv);
```

Reply:
I might not be understanding this correctly, but I don't think that is needed; the assignment operator is overridden here, which copies the inventory: https://github.com/minetest/minetest/blob/c63c05b141d994e801c82344499d86f5b67a2e3f/src/inventory.cpp#L854-L867

Reply:
This enables the inventory to be moved not copied. Might not work in this case if there's a custom copy-operator.